### PR TITLE
Faqs integration branch

### PR DESCRIPTION
**Pull request by ckeisato**
Tied to
https://github.com/CasperSleep/nightshade/pull/79

### CODE REVIEW

File: src/hero/hero.macros.html [hunk 1]
```diff
@@ -12,14 +12,14 @@
...
 {% macro hero(heading, content=null, background_path=null, featured=false, imgix_class='imgix-fluid--nocss', class=null, id=null) %}
 <header class="hero{% if featured %} hero--featured{% endif %} {{ imgix_class }}{% if class %} {{ class }}{% endif %}"{% if id %} id="{{ id }}"{% endif %}{% if background_path %} data-src="https://casper.imgix.net/{{ background_path }}"{% endif %}>
   <h1 class="hero-heading">{{ heading }}</h1>
   {% if content %}
   <h2 class="hero-subheading">{{ content }}</h2>
+  {% else %}
+  {{ caller() }}
   {% endif %}
 </header>
 {% endmacro %}
```
Comment (misscs) [inline src/hero/hero.macros.html:25]: @ckeisato / @josephco: 

This is a good time to set some standards for readability. Personally, I find the macro hard to parse without some whitespace and indentation. Here are some proposals, what do ya think? :) 

Option a: No whitespace, indent in control loops 

```
<header class="hero{% if featured %} hero--featured{% endif %} {{ imgix_class }}{% if class %} {{ class }}{% endif %}"{% if id %} id="{{ id }}"{% endif %}{% if background_path %} data-src="https://casper.imgix.net/{{ background_path }}"{% endif %}>
  <h1 class="hero-heading">{{ heading }}</h1>
  {% if content %}
    <h2 class="hero-subheading">{{ content }}</h2>
  {% else %}
    {{ caller() }}
  {% endif %}
</header>
```

Option b: Indent and whitespace

```
<header class="hero{% if featured %} hero--featured{% endif %} {{ imgix_class }}{% if class %} {{ class }}{% endif %}"{% if id %} id="{{ id }}"{% endif %}{% if background_path %} data-src="https://casper.imgix.net/{{ background_path }}"{% endif %}>
  <h1 class="hero-heading">{{ heading }}</h1>

  {% if content %}
    <h2 class="hero-subheading">{{ content }}</h2>
  {% else %}
    {{ caller() }}
  {% endif %}
</header>
```

File: src/links/links.macros.html [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+
+{#
+  * Renders a link with an icon to the right.
+  * @param {string} content - Link text content
+  * @param {string} href - Path or full URL for link
+  * @param {string} [icon=arrow] - Name of icon that renders and SVG icon
+  * @param {string} [class] - Class name for style or JS targeting
+  * @param {string} [id] - ID for JS targeting
+#}
+{% macro link(content, href, icon='arrow', class=null, id=null) %}
```
Comment (misscs) [inline src/links/links.macros.html]: whitespace between comment and macro

File: src/links/links.macros.html [hunk 3]
```diff
@@ -0,0 +1,25 @@
...
+<a class="link{% if icon %} link--icon{% endif %}{% if class %} {{ class }}{% endif %}"{% if id %} id="{{ id }}"{% endif %} href="{{ href }}">
+  {{ content }}
+  {% if icon %}
+    {{ icon_macro(icon) }}
```
Comment (misscs) [inline src/links/links.macros.html]: what do we think about a default? Will we always use the link macro with certain icon(s)? Weigh in from design here would be great!

File: src/links/_base.scss [hunk 4]
```diff
@@ -0,0 +1,9 @@
+/**
+ * @file Link
+ * @module Link
+ * @overview Link and top of page featured link modules
```
Comment (misscs) [inline src/links/_base.scss]: Not sure what page featured link modules means. Perhaps we can clarify this a bit so there is less ambiguity for devs.

File: src/links/links.macros.html [hunk 5]
```diff
@@ -19,7 +19,7 @@
 <a class="link{% if icon %} link--icon{% endif %}{% if class %} {{ class }}{% endif %}"{% if id %} id="{{ id }}"{% endif %} href="{{ href }}">
   {{ content }}
   {% if icon %}
-    {{ icon_macro(icon) }}
+    {{ icon_macro(icon, size="tiny") }}
   {% endif %}
 </a>
 {% endmacro %}
```
Comment (misscs) [inline src/links/links.macros.html:36]: Not a super big deal in this context, but worth noting so as well work on improving our commits that the update to macro doesn't fit the description of this commit. Perhaps could have been rebased/squashed into the commit the added the macro file. 

Overall though, :100: on small, single purpose commits. <3'ing it

File: src/links/links.macros.html [hunk 6]
```diff
@@ -8,18 +8,29 @@
...
+  * @param {string} content - Link text content
+  * @param {string} href - Path or full URL for link
+  * @param {string} [class] - Class name for style or JS targeting
+  * @param {string} [id] - ID for JS targeting
+#}
+{% macro link(content, href, class=null, id=null) %}
+<a class="link{% if class %} {{ class }}{% endif %}"{% if id %} id="{{ id }}"{% endif %} href="{{ href }}">
+  {{ content }}
+</a>
+{% endmacro %}
```
Comment (misscs) [inline src/links/links.macros.html:21]: I've found from past projects that having a granular macro such as link often created more headache than writing the code. We didn't have content in yaml though so that could be the problem. 

It is fine to keep this as sometimes using the macro make keep code more readable!

File: src/lists/lists.macros.html [hunk 7]
```diff
@@ -0,0 +1,34 @@
...
+    </li>
+    {% else %}
+    <li>
+      {{ link(item.content, href=item.href) }}
+    </li>
+    {% endif %}
+  {% endfor %}
+  </ul>
+</div>
+{% endmacro %}
```
Comment (misscs) [inline src/lists/lists.macros.html:36]: Since this pattern specifically is a list with a heading, being more explicit in our naming would make it clearer what this does. I would assume a list macro just makes a list.

File: src/lists/_base.scss [hunk 8]
```diff
@@ -5,5 +5,5 @@
  */
 
 .list {
-  padding: ms(1) 0;
+  margin: ms(1) 0;
```
Comment (misscs) [inline src/lists/_base.scss:8]: @josephco also brought this up, but we may just want to standardize on bottom or top that way we don't need to do any overrides for final elements. Maybe this isn't an issue for this module. Fit and finish could shed some more insight.

File: src/forms/_input_group.scss [hunk 9]
```diff
@@ -0,0 +1,63 @@
+// Input group has an input and a button_one
+// If the button has text, it is alongside the input field, if it has an icon, it is
+// positioned absolutely on top
+
```
Comment (misscs) [inline src/forms/_input_group.scss:7]: We'll want to continue the JSDoc style docs at the top of the page. This description is great for @overview directive. One fix would be to just use `button` instead of `button_one`

File: src/forms/_input_group.scss [hunk 10]
```diff
@@ -0,0 +1,63 @@
...
+.input-group {
+  display: flex;
+
+  .input {
+    @extend %body--sans-small;
```
Comment (misscs) [inline src/forms/_input_group.scss:12]: Extending this font style feels fragile. curious what @josephco thinks.

File: src/forms/forms.macros.html [hunk 11]
```diff
@@ -24,3 +24,21 @@
...
+  * @param {string} [button_icon] - Icon name for Buttons with icons
+  * @param {string} [class] - Class name for style or JS targeting
+  * @param {string} [id] - ID for JS targeting
+#}
+{% macro input_button(input_name, input_placeholder=null, button_content='', button_style='primary', button_icon=null, class=null, id=null) %}
+<div class="input-group {% if button_icon %}input-group--button-icon{% else %}input-group--button{% endif %}{% if class %} {{ class }}{% endif %}" {% if id %} id="{{ id }}"{% endif %}>
+  <input class="input" name="{{ input_name }}" placeholder="{{ input_placeholder }}" required>
+  {{ button(button_content, style=button_style, icon=button_icon, type="submit") }}
+</div>
+{% endmacro %}
```
Comment (misscs) [inline src/forms/forms.macros.html:46]: I find myself going back and forth on this pattern. From one end it feels too abstract and generic, like it accomplishing too many things. But from another end, it feels like it is robust to handle different situations. 

There's a part of me that would prefer to have two explicit patterns: One for search input and the other for an input with a button (newsletter and promo fields on our current site).

File: src/forms/forms.macros.html [hunk 12]
```diff
@@ -24,3 +24,21 @@
...
+  * @param {string} [input_placeholder] - Placeholder text for the input
+  * @param {string} [button_content=''] - Button text content
+  * @param {string} [button_style=primary] - Button style
+  * @param {string} [button_icon] - Icon name for Buttons with icons
+  * @param {string} [class] - Class name for style or JS targeting
+  * @param {string} [id] - ID for JS targeting
+#}
+{% macro input_button(input_name, input_placeholder=null, button_content='', button_style='primary', button_icon=null, class=null, id=null) %}
+<div class="input-group {% if button_icon %}input-group--button-icon{% else %}input-group--button{% endif %}{% if class %} {{ class }}{% endif %}" {% if id %} id="{{ id }}"{% endif %}>
+  <input class="input" name="{{ input_name }}" placeholder="{{ input_placeholder }}" required>
```
Comment (misscs) [inline src/forms/forms.macros.html]: Need to add `type` and make `required` a boolean. Sometimes the input may not be required

File: src/forms/_input_group.scss [hunk 13]
```diff
@@ -0,0 +1,63 @@
...
+  [data-button] {
+    height: 48px;
+  }
+}
+
+.input-group--button-icon {
+  position: relative;
+
+  .input {
+    padding-right: 60px;
```
Comment (misscs) [inline src/forms/_input_group.scss:51]: Feels like a magic number!

File: src/links/links.macros.html (hunk 6 above)
Comment (ckeisato) [inline src/links/links.macros.html:21]: I'm ok with not having a link macro, would all link styling then be a part of typography?

File: src/hero/hero.macros.html (hunk 1 above)
Comment (ckeisato) [inline src/hero/hero.macros.html:25]: Do you want new lines for control loops or between elements?  Or both?  I'm fine either way but if we're going to have more whitespace in the macro, I think @josephco was saying we have to strip the output (which we should probably do anyway).

File: src/lists/lists.macros.html (hunk 7 above)
Comment (ckeisato) [inline src/lists/lists.macros.html:36]: Sure that makes sense, will be amending this to be `list--heading`.

File: src/forms/forms.macros.html (hunk 11 above)
Comment (ckeisato) [inline src/forms/forms.macros.html:46]: Isn't the search input also an input with a button?  Or do you mean a input with an obvious button (like email sign up) vs an input with a ghost button so that it looks like it's inside the input?

File: src/lists/lists.macros.html (hunk 7 above)
Comment (ckeisato) [inline src/lists/lists.macros.html:36]: @misscs Since we may have lists with or without headings, would it make sense to keep it called `List` but have the heading be optional?
